content/blog/improved-kubernetes-await-logic/index.md

```diff
@@ -0,0 +1,293 @@
+   For example this will wait until the resource has a "Synced=True" condition:
+
+       "pulumi.com/waitFor": "condition=Synced"
+
+   and this will wait until it has a "Reconciling=False" condition:
+
+       "pulumi.com/waitFor": "condition=Reconciling=False"
+
+3. Lastly, the `pulumi.com/waitFor` annotation also accepts a JSON-encoded string containing an array of one or more "jsonpath=" or "condition=" expressions.
+   For example, this resource will be considered ready when it simultaneously has a "Bar=True" condition and a populated "foo" field.
+
+       "pulumi.com/waitFor": '["jsonpath={.foo}", "condition=Bar"]'
+
+### Custom readiness example: cert-manager
+
+[cert-manager](http://cert-manager.io) is a popular X.509 certificate management solution with some installation nuances.
+If you follow the default [instructions](https://cert-manager.io/docs/installation/) with a goal of also creating a new Certificate as part of your Pulumi stack, you might end up with something like this:
+
+```typescript
+import * as kubernetes from "@pulumi/kubernetes";
+
+const install = new kubernetes.yaml.v2.ConfigFile("install",
+  {
+    file: "https://github.com/cert-manager/cert-manager/releases/download/v1.15.2/cert-manager.yaml",
+  },
+);
+
+const issuer = new kubernetes.apiextensions.CustomResource("issuer",
+  {
+    apiVersion: "cert-manager.io/v1",
+    kind: "ClusterIssuer",
+    metadata: {
+      name: "selfsigned-issuer",
+    },
+    spec: {
+      selfSigned: {},
+    },
+  },
+  { dependsOn: [install] }
+);
+
+new kubernetes.apiextensions.CustomResource("certificate", {
+  apiVersion: "cert-manager.io/v1",
+  kind: "Certificate",
+  metadata: {
+    name: "selfsigned-cert",
+  },
+  spec: {
+    isCA: true,
+    commonName: "my-ca",
+    secretName: "root-secret",
+    privateKey: {
+      algorithm: "ECDSA",
+      size: 256,
+    },
+    issuerRef: {
+      name: issuer.metadata.name,
+      kind: issuer.kind,
+      group: "cert-manager.io",
+    },
+  },
+});
+```
+
+This uses a [v2 ConfigFile](https://www.pulumi.com/blog/kubernetes-yaml-v2/) to `kubectl apply` the cert-manager custom resource definitions (CRDs), and then it creates a self-signed Issuer and Certificate using those CRDs.
+
+The dependencies between resources _appear_ reasonable, however this program will likely fail when it's first run:
+
+{{% notes type="warning" %}}
+error: resource "kubernetes:cert-manager.io/v1:ClusterIssuer::issuer" was not successfully created by the Kubernetes API server: Internal error occurred: failed calling webhook "webhook.cert-manager.io": failed to call webhook: Post "https://cert-manager-webhook.cert-manager.svc:443/validate?timeout=30s": tls: failed to verify certificate: x509: certificate signed by unknown authority
+{{% /notes %}}
+
+We see this "certificate signed by unknown authority" error because a few things need to happen before the cert-manager web hooks are usable:
```

Comment:
```suggestion
We see this "certificate signed by unknown authority" error because a few things need to happen before the cert-manager webhooks are operational:
```